Implementation of DisplayEvent

abudi511 opened the pull request:
Implemented all of the Display Events except for the view.
Made the interactor so that it takes an event and makes it readable to the user. 
The following are the files that have changed:
- DisplayEventPresenter
- DisplayEventState
- DisplayEventViewModel
- DisplayEventInteractor
- DisplayEventOutputData
- EventView (not yet completed)

abudi511 committed:
.

hugo-ye reviewed (changes requested):
Great work Abdullah! There are somethings I would like to address int the code. You should improve your implementation of ViewModel to better adhere to inheritance rules. You passed the generic `<Object>` as the state to your `DisplayEventViewModel`. Instead, you should pass your `DisplayEventState`. This way you do not have to redefine the State in your ViwModel.

`DisplayEventState`
Here `PropertyChangeSupport` is misplaced. Moreover, you do not need to define a constructor. The state variables are not set when the state is constructed but instead by the ViewModel using the setters.

Hence, remove the constructor, the field `support`, the method `addPropertyChangeListener`.

Now for `DisplayEventViewModel`,
Remove all the fields and methods. Then add `super("")`, to instantiate the parent class ViewModel, and pass in the ViewName of the particular view associated with your ViewModel. Then set the state by calling setState(""), with a new state of yours.

Please fix this, and I will give you more feedback on the rest. The code in general looks good!

abudi511 committed:
passed in DisplayEventState as the state to DisplayEventViewModel

for the DisplayEventViewModel i removed all the methods and fields and called super

abudi511 committed:
passed in DisplayEventState as the state to DisplayEventViewModel

for the DisplayEventViewModel i removed all the methods and fields and called super

hugo-ye commented:
Great work. The State looks very good now. I would like you to address this in ViewModel.
The view it is tied to is the Event details view. On a different branch this has actually been declared in the Eventview.java, as follows.

`public final String viewName = "event details";`

I would like you to change your viewmodel so that it doesn't pass in super(viewname) where you injected a new viewName in the constructor. Rather, just write "event details". This way the ViewModel is tied to the correct view.

abudi511 committed:
changed the viewModel so that super takes in the string "event details"

abudi511 commented:
done

hugo-ye commented:
In DisplayEventInteractor, you ineject a DataAccessInterface. You do not use it. Displaying an event does not go through the persistent storage at all. That is because you do not need it. The redundant code should be removed.

abudi511 committed:
removed the DAO in the interactor

abudi511 committed:
Merge branch 'main' into DisplayEvent_Implementation

hugo-ye commented:
One other thing. I noticed that you have not removed the redundant functions in your ViewModel. Please complete that.

hugo-ye commented:
Edit to my previous message. You seem to have lots of code remnant in the form of comments. Please remove them before we can merge it to main, for code style and cleanliness.

abudi511 committed:
Update DisplayEventState.java

removed the comment

abudi511 committed:
Update DisplayEventViewModel.java

removed the comments

abudi511 committed:
Update DisplayEventInteractor.java

fixed the CA

abudi511 committed:
Update DisplayEventOutputData.java

fixed CA

abudi511 committed:
Update DisplayEventPresenter.java

fixed CA

abudi511 committed:
Update DisplayEventPresenter.java

hugo-ye reviewed (approved):
Good work. You've addressed all of my concerns regarding the feature. For the final product though, in the code style polishing phase, remember to remove some commented code, and improve on style.

Amir8-5 reviewed (approved):
Looks good, good job
